qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayTestRepositoryCloud.java

```diff
@@ -0,0 +1,137 @@
+                    total
+                    limit
+                    results {
+                      issueId
+                      testType {
+                        name
+                        kind
+                      }
+                      steps {
+                        id
+                        data
+                        action
+                        result
+                      }
+                    }
+                  }
+                }
+                """;
+        // For big test sets, we need to send multiple queries, since JQL queries that would return more than
+        // GRAPH_QL_QUERY_ISSUES_LIMIT issues result in error responses (see link above).
+        Map<String, XrayTestStepResponseDto[]> stepsByIssue = new HashMap<>();
+        for (int n = 0; n < testIssueKeys.size(); n = n + GRAPH_QL_QUERY_ISSUES_LIMIT) {
+            List<String> subset = testIssueKeys.stream()
+                    .skip(n)
+                    .limit(GRAPH_QL_QUERY_ISSUES_LIMIT)
+                    .toList();
+            String jql = String.format("issue in (%s)", String.join(",", subset));
+            GraphQLRequestDto dto = new GraphQLRequestDto();
+            dto.setQuery(query);
+            dto.addVariable("jql", jql);
+            dto.addVariable("limit", GRAPH_QL_QUERY_ISSUES_LIMIT);
+            RequestBuilder request = WebService.buildRequest(getURIGraphQL());
+            request.getBuilder()
+                    .header(HttpHeaders.AUTHORIZATION, getXrayAuthorizationHeaderValue())
+                    .accept(MediaType.APPLICATION_JSON_TYPE);
+            String requestData = GsonFactory.getInstance().toJson(dto);
+            try (Response response = WebService.post(request, GsonFactory.getInstance().toJsonTree(dto))) {
+                String responseData = response.readEntity(String.class);
+                if (response.getStatus() == Response.Status.OK.getStatusCode()) {
+                    stepsByIssue.putAll(parseTestStepResponse(requestData, responseData));
+                } else {
+                    String reason = String.format(
+                            "%d %s: %s",
+                            response.getStatus(),
+                            response.getStatusInfo().getReasonPhrase(),
+                            responseData
+                    );
+                    QtafFactory.getLogger().error(
+                            String.format(
+                                    "[QTAF Xray Plugin] Failed to get test steps for issues using query '%s': %s",
+                                    requestData,
+                                    reason
+                            )
+                    );
+                    break;
+                }
+            }
+        }
+        // Unfortunately, Xray only returns issue IDs instead of issue keys. Therefore, we need this extra step now.
+        return convertIssueIdsToIssueKeys(stepsByIssue, testIssueKeys);
+    }
+
```

Comment:
Please add DocString

Reply:
Fixed in [8531207](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/85312073da996cc59b63375b5721a22ae2ebcfa7).